virttest/qemu_vm.py

```diff
@@ -1632,83 +1535,47 @@ def sort_key(dev):
                 devices.insert(pvpanic_dev)
 
         # Add serial console redirection
-        for serial in params.objects("serials"):
+        serials = params.objects('serials')
+        if serials:
+            self.serial_session_device = serials[0]
+            host = params.get('serial_chardev_host', '127.0.0.1')
+            free_ports = utils_misc.find_free_ports(
+                5000, 6000, len(serials), host)
+            reg_count = 0
+        self.virtio_ports = []
+        for index, serial in enumerate(serials):
             serial_filename = vm.get_serial_console_filename(serial)
-            cmd = add_serial(devices, serial, serial_filename)
-            devices.insert(StrDev('SER-%s' % serial, cmdline=cmd))
+            serial_params = params.object_params(serial)
+            # Workaround for console issue, details:
+            # http://lists.gnu.org/archive/html/qemu-ppc/2013-10/msg00129.html
+            if 'ppc' in params.get('vm_arch_name', arch.ARCH)\
+                    and serial_params.get('serial_type') == 'spapr-vty':
+                reg = 0x30000000 + 0x1000 * reg_count
+                serial_params['serial_chardev_reg'] = hex(reg)
```

Comment:
`chardev_reg`

Reply:
@yanan-fu Updated, but this one is `serial_reg`, it's an option for serial device, not chardev.